### PR TITLE
Pr 16

### PR DESCRIPTION
**Pull request by grostim**

### CODE REVIEW

File: pdfbourso/pdfbourso.py [hunk 1]
```diff
@@ -68,9 +74,9 @@ class PDFBourso(beangulp.Importer):
...
 
-    REGEX_OPCVM_MONTANT = r"Montant brut\s*Droits d'entrée\s*Frais H.T.\s*T.V.A.\s*Montant net au débit de votre compte\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s"
-    REGEX_OPCVM_DETAILS = r"(\d{1,2}\/\d{2}\/\d{4})\s*(\d{0,3}\s\d{1,3}[.,]?\d{0,4})\s*([\s\S]{0,20})?\s*"
-    REGEX_OPCVM_COURS = r"Valeur liquidative :\s*(\d{0,3}\s\d{1,3}[,.]\d{0,4})\s([A-Z]{1,3})"
+    REGEX_OPCVM_MONTANT = r"Montant brut\s*Droits (?:d'entrée|de sortie)\s*Frais H.T.\s*T.V.A.\s*Montant net au (?:débit|crédit) de votre compte\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(?:(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*)?(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s"
+    REGEX_OPCVM_DETAILS = r"Date\s*Quantité\s*Informations sur la valeur\s*Informations sur l'exécution[\s\S]*?(\d{1,2}\/\d{2}\/\d{4})\s*(\d+(?:\s\d{3})*(?:[,.]\d{1,4})?)\s*([\s\S]{0,80}?)\s*Référence\s*:"
```
Comment (grostim) [inline pdfbourso/pdfbourso.py:78]: 🟡 **P1 — Ancrage « Référence : » fragile**

Le nouveau `REGEX_OPCVM_DETAILS` utilise `Référence\s*:` comme marqueur de fin. Si Boursorama modifie ce libellé (majuscules, abréviation « Réf : »), l'extraction échoue silencieusement.

**Suggestion :** rendre le suffixe plus tolérant :
```python
# Avant
REGEX_OPCVM_DETAILS = r"...Référence\s*:"
# Après
REGEX_OPCVM_DETAILS = r"...Réf(?:érence)?\s*:"
```

L'alternative garde la spécificité tout en tolérant « Réf : » et les variantes de casse si on ajoute `re.IGNORECASE`.

Comment (grostim) [inline pdfbourso/pdfbourso.py:78]: 🔵 **P2 — Capture payee bornée à 80 caractères**

`([\s\S]{0,80}?)` tronque silencieusement les noms de fonds dépassant 80 caractères. Certains noms OPCVM peuvent être plus longs.

**Suggestion :** capture non bornée, le suffixe « Référence : » empêche le runaway matching :
```python
# Avant
([\s\S]{0,80}?)\s*Référence\s*:
# Après
([\s\S]*?)\s*Référence\s*:
```

File: pdfbourso/pdfbourso.py [hunk 2]
```diff
@@ -68,9 +74,9 @@ class PDFBourso(beangulp.Importer):
 
     REGEX_ACTION_MONTANT = r"Montant brut\s*Commission\s*Frais\s\(.\)\s*Montant net au crédit de votre compte\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(?:(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3}))?\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*"
 
-    REGEX_OPCVM_MONTANT = r"Montant brut\s*Droits d'entrée\s*Frais H.T.\s*T.V.A.\s*Montant net au débit de votre compte\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s"
-    REGEX_OPCVM_DETAILS = r"(\d{1,2}\/\d{2}\/\d{4})\s*(\d{0,3}\s\d{1,3}[.,]?\d{0,4})\s*([\s\S]{0,20})?\s*"
-    REGEX_OPCVM_COURS = r"Valeur liquidative :\s*(\d{0,3}\s\d{1,3}[,.]\d{0,4})\s([A-Z]{1,3})"
+    REGEX_OPCVM_MONTANT = r"Montant brut\s*Droits (?:d'entrée|de sortie)\s*Frais H.T.\s*T.V.A.\s*Montant net au (?:débit|crédit) de votre compte\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*(?:(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s*)?(\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s([A-Z]{3})\s"
```
Comment (grostim) [inline pdfbourso/pdfbourso.py:77]: 🔵 **P2 — Groupes numériques fragiles après ajout de TVA**

L'ajout de la colonne TVA dans `REGEX_OPCVM_MONTANT` décale les indices de Montant Net de 7/8 à 9/10. Si Boursorama intercale une future colonne avant « Montant net », les indices casseront sans erreur visible.

**Suggestion :** utiliser des groupes nommés pour les champs critiques :
```python
REGEX_OPCVM_MONTANT = r"""Montant brut\s*Droits (?:d'entrée|de sortie)\s*Frais H\.T\.\s*T\.V\.A\.\s*Montant net au (?:débit|crédit) de votre compte\s*(?P<montant_net>\d{0,3}\s*\d{1,3}[,.]\d{1,3})\s(?P<currency_net>[A-Z]{3})\s*"""
```

Ce n'est pas bloquant aujourd'hui, mais le coût de migration augmentera avec chaque nouvelle colonne ajoutée.

File: pdfbourso/pdfbourso.py [hunk 3]
```diff
@@ -177,11 +183,18 @@ def account(self, file):
...
                     self._debug(f"Compte et ISIN : {self.accountList[compte]}:{isin}")
                     return f"{self.accountList[compte]}:{isin}"
-            elif self.type in ["DividendeBourse", "EspeceDividende"]:
+            elif self.type in ["DividendeBourse", "EspeceBourse", "EspeceDividende"]:
                 return f"{self.accountList[compte]}:Cash"
             else:
                 return self.accountList[compte]
 
+    def _resolve_cash_statement_account(self, compte: str, text: str) -> str:
+        """Return the posting/balance account for cash statement-like documents."""
```
Comment (grostim) [inline pdfbourso/pdfbourso.py]: 🔵 **P2 — Docstring en anglais**

Le reste du code utilise des docstrings en français. Cohérence.

**Suggestion :**
```python
def _resolve_cash_statement_account(self, compte: str, text: str) -> str:
    """Résout le compte de posting/balance pour les relevés de type compte-espèces."""
```

File: pdfbourso/pdfbourso.py [hunk 4]
```diff
@@ -741,11 +756,14 @@ def _extract_compte(self, file, text, document):
...
         if match:
-            compte = match.group(0).split(" ")[-1]
+            compte = match.group(1).strip()
 
         # Si debogage, affichage de l'extraction
         self._debug(f"Numéro de compte extrait : {compte}")
         columns = self._find_compte_columns(text)
+        base_account = self.accountList[compte]
+        statement_account = self._resolve_cash_statement_account(compte, text)
+        balances_only_statement = base_account in self.BOURSE_CASH_PARENT_ACCOUNTS
```
Comment (grostim) [inline pdfbourso/pdfbourso.py]: 🟡 **P1 — Logique redondante avec `_resolve_cash_statement_account()`**

`_resolve_cash_statement_account()` (l. 191-196) et `balances_only_statement` (l. 766) évaluent la même condition `base_account in self.BOURSE_CASH_PARENT_ACCOUNTS` indépendamment. Si la liste des comptes concernés évolue, les deux endroits doivent être mis à jour.

**Suggestion :** dériver `balances_only_statement` du résultat de `_resolve_cash_statement_account()` pour une source unique de vérité :
```python
statement_account = self._resolve_cash_statement_account(compte, text)
balances_only_statement = statement_account != self.accountList[compte]
```

Le booléen est alors vrai quand `:Cash` a été ajouté — pas besoin d'une vérification séparée contre `BOURSE_CASH_PARENT_ACCOUNTS`.

**Review by grostim — COMMENTED** [review]
## 🔍 Hermes Code Review — PR #17

**Verdict : Comment** — Pas de bug bloquant, améliorations suggérées.

### Résumé des changements
- Support des relevés OPCVM « Reprise » (crédit) : regex `REGEX_OPCVM_MONTANT`, `REGEX_OPCVM_DETAILS`, `REGEX_OPCVM_COURS` adaptés
- Correction du traitement TVA dans les opérations OPCVM (frais = Frais HT + Droits + TVA)
- Gestion des séparateurs de milliers dans les quantités et VL
- Comptes espèces PEA/CTO/PEAPME : distinction compte titre vs compte cash pour les relevés « Compte »
- Simplification : `account()` attache `:Cash` automatiquement pour `EspeceBourse`

### 🔴 P0 — Aucun

### 🟡 P1 — 2 warnings (voir inline)
- **`REGEX_OPCVM_DETAILS`** : ancrage « Référence : » fragile — `Réf(?:érence)?\s*:` plus tolérant
- **Logique redondante** entre `_resolve_cash_statement_account()` et `balances_only_statement` — même condition évaluée 2×, peut être consolidée

### 🔵 P2 — 3 suggestions (voir inline)
- Groupes nommés pour `REGEX_OPCVM_MONTANT` (résilience aux changements de colonnes)
- Capture payee non bornée dans `REGEX_OPCVM_DETAILS` plutôt que `{0,80}`
- Docstring anglais → français (cohérence)

### ✅ Ce qui est bon
- Tests propres et bien ciblés (reprise crédit, bruit PDF, séparateurs milliers, TVA)
- Aucune régression : `account()` simplifie `_extract_espece_bourse()` proprement
- Les 9 tests passent, aucun secret, aucun souci de sécurité

---
*Reviewed by Hermes Agent*

**Review by grostim — COMMENTED** [review]
## 🔍 Hermes Code Review — PR #17 (2ᵉ passe)

**Verdict : Comment** — Le code est correct, les tests passent. Quelques points dattention.

### ✅ Revue précédente — statut
- ~~Docstring en anglais~~ → corrigé ✓
- ~~Logique redondante~~ → corrigé ✓
- 🟡 P1 — Ancrage Référence : fragile → encore présent
- 🔵 P2 — Capture payee 80 chars → encore présent
- 🔵 P2 — Groupes numériques fragiles → encore présent

### 🆕 Nouveaux findings
- 🟡 P1 — Comportement balances_only_statement non documenté (l. 834)
- 🔵 P2 — Paramètre text inutilisé dans _resolve_cash_statement_account
- 🔵 P2 — Paramètre monkeypatch inutilisé dans le test PEA

### ✅ Ce qui est propre
- Séparation commission / impôts dans _extract_dividende_bourse
- Support séparateurs de milliers dans REGEX_OPCVM_COURS
- Gestion TVA absent (fallback or 0.0)
- Tests de bruit PDF — bonne couverture défensive

---

File: pdfbourso/pdfbourso.py [hunk 5]
```diff
@@ -330,27 +335,54 @@ def _extract_dividende_bourse(self, file, text, document):
...
             for chunk in chunks:
                 try:
+                    gross = self._parse_decimal(chunk[5])       # montant brut global
+                    net = self._parse_decimal(chunk[7])         # net client
+                    commission = self._parse_decimal(chunk[6] or '0')  # commission TTC
+
                     postings = [
-                        self._create_posting("Revenus:Dividendes", self._parse_decimal(chunk[4]) * -1, "EUR"),
-                        self._create_posting("Depenses:Impots:IR", self._parse_decimal(chunk[5] or '0') + self._parse_decimal(chunk[6]), "EUR"),
-                        self._create_posting(compte, self._parse_decimal(chunk[7]), "EUR")
```
Comment (grostim) [inline pdfbourso/pdfbourso.py:339]: 🔵 **P2 — Paramètre `text` inutilisé.**

`_resolve_cash_statement_account(self, compte: str, text: str)` accepte `text` mais ne l'utilise pas dans le corps de la fonction. Si c'est intentionnel (usage futur), un commentaire l'expliquant évitera les questions. Sinon, retirer le paramètre.

**Review by grostim — COMMENTED** [review]
## 🔍 Commentaires additionnels (Hermes Agent — passe 2)

### 🟡 P1 — Comportement `balances_only_statement` non documenté (pdfbourso.py:834)

Le `if not balances_only_statement:` supprime silencieusement toutes les transactions pour les comptes PEA/PEAPME/CTO. C'est architecturalement correct (les transactions viennent de `_extract_espece_bourse`), mais aucun commentaire ne l'explique. Un futur mainteneur pourrait croire à un bug.

**Suggestion :** ajouter un commentaire avant le `if` :
```python
# Les relevés Compte pour PEA/PEAPME/CTO ne contiennent que des
# lignes de solde. Les transactions sont extraites séparément.
if not balances_only_statement:
```

### 🔵 P2 — Paramètre `monkeypatch` inutilisé (pdfbourso_test.py:108)

`test_extract_compte_pea_cash_statement_targets_pea_cash` accepte `monkeypatch` mais ne l'utilise pas. `_extract_compte` reçoit `text` directement.

---
*Reviewed by Hermes Agent*

**Review by grostim — COMMENTED** [review]
## 🔍 Hermes Code Review — PR #17 (sync)

**Verdict : Comment** — Le fix est partiel : 2/5 commentaires de la review précédente adressés. 3 restent ouverts.

### 🟡 P1 — 1 warning
- **`REGEX_OPCVM_DETAILS`** : ancrage « Référence : » toujours fragile (non corrigé malgré le commit message)

### 🔵 P2 — 2 suggestions
- **`REGEX_OPCVM_DETAILS`** : capture payee toujours bornée à 80 caractères
- **`_resolve_cash_statement_account()`** : paramètre `text` non utilisé

### ✅ Ce qui a été corrigé
- Logique redondante `balances_only_statement` : dérivée de `statement_account != self.accountList[compte]` ✓
- Docstring `_resolve_cash_statement_account` : anglais → français ✓
- Les 9 tests passent, aucune régression

---
*Reviewed by Hermes Agent*

**Comment by grostim** [comment]
🟡 **P1 — Ancrage « Référence : » toujours fragile**

Le commit message dit « resolve P1/P2 review comments » mais ce point n'a pas été corrigé. `Référence\s*:` dans `REGEX_OPCVM_DETAILS` (ligne ~78) reste inchangé.

**Suggestion :**
```python
# Avant
REGEX_OPCVM_DETAILS = r"...Référence\s*:"
# Après
REGEX_OPCVM_DETAILS = r"...Réf(?:érence)?\s*:"
```

Rendre le suffixe plus tolérant évite les échecs silencieux si Boursorama change la casse ou abrège le libellé.

**Comment by grostim** [comment]
🔵 **P2 — Capture payee toujours bornée à 80 caractères**

`([\s\S]{0,80}?)` dans `REGEX_OPCVM_DETAILS` (ligne ~78) tronque silencieusement les noms de fonds longs. Le suffixe « Référence : » empêche déjà le runaway matching.

**Suggestion :** 
```python
# Avant
([\s\S]{0,80}?)\s*Référence\s*:
# Après
([\s\S]*?)\s*Réf(?:érence)?\s*:
```

Capture non bornée, plus robuste aux noms longs.

**Comment by grostim** [comment]
🔵 **P2 — Paramètre `text` non utilisé dans `_resolve_cash_statement_account()`**

La signature `_resolve_cash_statement_account(self, compte: str, text: str)` (ligne ~191) reçoit `text` mais ne l'utilise pas. Tous les appelants passent ce paramètre inutilement.

**Suggestion :** soit le supprimer de la signature, soit le documenter (extension future ?).

**Comment by grostim** [comment]
## 🔍 Hermes Post-Merge Code Review

**PR:** #17 — Pr 16 (merged)
**Author:** @grostim
**Files changed:** 2 (+283 -73)
**Tests:** 9/9 passed ✅

**Verdict:** plusieurs P1 non résolus de la review précédente, plus quelques nouveaux problèmes. Rien de critique — le cœur de la PR (correction dividende brut, TVA OPCVM, PEA:Cash) est solide.

### ⚠️ Warnings (P1)

- **`pdfbourso.py:78` — Ancrage « Référence : » toujours fragile** — déjà flaggé dans la review initiale. Un espace insécable ou une abréviation « Réf : » casse l'extraction. Changer en `Réf(?:érence)?\s*:` avec `re.IGNORECASE`.
- **`pdfbourso.py:834` — `balances_only_statement` ignore TOUTES les transactions** — pour les relevés PEA/PEAPME/CTO, les virements et prélèvements hors-trading (ex: `VIR CTo→PEA`) sont perdus. Soit documenter ce choix, soit filtrer uniquement les transactions déjà couvertes par les documents individuels (ACHAT/VENTE).
- **`pdfbourso.py:180` — `EspeceDividende` est un type mort** — `DOCUMENT_TYPES` ne le définit pas. Cette branche `elif` n'est jamais atteinte. (Préexistant — à nettoyer.)

### 💡 Suggestions (P2)

- **`pdfbourso.py:79` — Payee OPCVM tronqué à 80 car.** — `{0,80}?` coupe les noms longs de fonds. Remplacer par `([\s\S]*?)` (le suffixe `Référence` empêche le runaway). Déjà flaggé, pas fixé.
- **`pdfbourso.py:191` — Paramètre `text` inutilisé** dans `_resolve_cash_statement_account()`. Supprimer si non prévu.
- **`pdfbourso.py:77` — Indices numériques fragiles** — l'ajout TVA décale tous les groupes. Groupes nommés `(?P<montant_net>...)` recommandés. Déjà flaggé, pas fixé.

### ✅ Bien vu

- La correction du dividende (montant brut au lieu du montant unitaire) — le bug était réel et bien résolu.
- Tests additionnels pour OPCVM reprise crédit avec bruit de mise en page, séparateur de milliers, TVA présente — couverture réelle de edge cases PDF.
- Le pattern `_resolve_cash_statement_account` / `balances_only_statement` est propre et évite la duplication Cash:Cash.

---
*Reviewed by Hermes Agent (post-merge)*

**Comment by grostim** [comment]
### Inline comments (API refuses line resolution on merged PR — posted here instead)

---

**🟡 P1 — `pdfbourso.py:186` : `EspeceDividende` est un type mort**

`DOCUMENT_TYPES` ne définit pas `EspeceDividende` — cette branche `elif` n'est jamais atteinte. (Préexistant, à nettoyer.)

**Fix :** soit retirer `EspeceDividende` des conditions, soit l'ajouter à `DOCUMENT_TYPES`.

---

**🟡 P1 — `pdfbourso.py:834` : `balances_only_statement` ignore les virements/prélèvements non-boursiers**

Quand le relevé Compte est un compte-espèces PEA/PEAPME/CTO, TOUTES les transactions sont sautées. Les virements (VIR CTo→PEA) et prélèvements sans document individuel sont perdus.

**Fix :** soit documenter ce choix, soit filtrer uniquement les transactions déjà couvertes par les docs individuels.

---

**🔵 P2 — `pdfbourso.py:191` : Paramètre `text` inutilisé**

```python
def _resolve_cash_statement_account(self, compte: str, text: str) -> str:
```

`text` n'est jamais utilisé dans le corps. Si pas prévu → retirer.

---

**🔵 P2 — `pdfbourso.py:77` : Indices numériques fragiles** (déjà flaggé, pas fixé)

L'ajout TVA décale les groupes de Montant Net de 7/8 à 9/10. Groupes nommés `(?P<montant_net>...)` recommandés.
